railties/test/railties/engine_test.rb

```diff
@@ -252,6 +254,38 @@ def index
       assert_equal "Hello bukkits\n", response[2].body
     end
 
+    test "does not eager load controller actions in development" do
+      @plugin.write "app/controllers/bukkit_controller.rb", <<-RUBY
+        class BukkitController < ActionController::Base
+          def index;end
+          def show;end
+        end
+      RUBY
+
+      boot_rails
+
+      assert_called(BukkitController, :internal_methods, "#action_methods is not memoized", returns: []) do
+        BukkitController.action_methods
+      end
+    end
+
+    test "eager loads controller actions when eager_load is enabled" do
+      add_to_env_config("development", "config.eager_load = true")
+
+      @plugin.write "app/controllers/bukkit_controller.rb", <<-RUBY
+        class BukkitController < ActionController::Base
+          def index;end
+          def show;end
+        end
+      RUBY
+
+      boot_rails
+
+      assert_not_called(BukkitController, :internal_methods, "#action_methods is already memoized") do
```

Comment:
Another way to test it would be checking the memoized instance variable:

`assert BukkitController.instance_variable_get(:@action_methods)`

But rather I prefered to check that the depending methods are not invoked.